### ISSUE TITLE
The eslintrc should be overhauled for explicit Node v4 support

### ISSUE THREAD

**Issue by appden** [rnpm/rnpm-plugin-link#93]
It's currently set to use `babel-eslint`, which is allowing destructuring and other features that Node v4 (and even v5) do not support.

**Comment by Kureev** [comment]
We need to use the same configure as react-native. We already merged rnpm into RN core, so I don't see a point of changing RNPM repo now. I think you'll be able to use some commands in the next release already 😉

**Comment by appden** [comment]
Respectfully I disagree. If all the JS code for `rnpm` needs to be Node v4 compatible, then it should have an `eslintrc` that overrides the root react-native rules to restrict the acceptable syntax to what will work in Node v4. Nested `eslintrc` files will override ones in parent directories, and you could even use `root: true` to set it as the root rule set.

P.S. Very excited about `rnpm` being merged into React Native. Congrats!

**Comment by Kureev** [comment]
Well, I think you're mostly correct. The only thing is that [react-native requires node v4+ as well](https://github.com/facebook/react-native/blob/master/package.json#L11), so I though mb it can fit for rnpm too, because we don't want to introduce yet another code style in subfolder

**Comment by appden** [comment]
I agree about the code style, that's why I think the `eslintrc` should just disable ES6 features that are known not to be supported by Node v4, but not mess with style rules inherited from the parent.

**Comment by Kureev** [comment]
Ok, sounds reasonable 👍
